src/components/POLine/Location/LocationForm.js

```diff
@@ -27,33 +33,53 @@ const validateNotNegative = (value) => (value < 0
   ? <FormattedMessage id="ui-orders.validation.quantity.canNotBeLess0" />
   : undefined);
 
-const validateQuantityPhysical = (value, { cost, locations = [] }) => {
+const validateQuantityPhysical = (value, { cost, locations = [], physical, orderFormat }) => {
   const allLocationsQuantity = getTotalLocationsQuantities(locations, 'quantityPhysical');
-  const overallLineQuantity = get(cost, 'quantityPhysical', 0);
+  const overallLineQuantity = Number(get(cost, 'quantityPhysical', 0));
+  const inventoryType = get(physical, 'createInventory', '');
 
-  return allLocationsQuantity === overallLineQuantity
-    ? undefined
-    : <FormattedMessage id="ui-orders.location.quantityPhysical.notMatch" />;
+  return isLocationPhysicalQuantityRequired(orderFormat, inventoryType)
+  && allLocationsQuantity !== overallLineQuantity
+    ? <FormattedMessage id="ui-orders.location.quantityPhysical.notMatch" />
+    : undefined;
 };
 
-const validateQuantityElectronic = (value, { cost, locations = [] }) => {
+const validateQuantityElectronic = (value, { cost, locations = [], eresource, orderFormat }) => {
   const allLocationsQuantity = getTotalLocationsQuantities(locations, 'quantityElectronic');
-  const overallLineQuantity = get(cost, 'quantityElectronic', 0);
+  const overallLineQuantity = Number(get(cost, 'quantityElectronic', 0));
+  const inventoryType = get(eresource, 'createInventory', '');
 
-  return allLocationsQuantity === overallLineQuantity
-    ? undefined
-    : <FormattedMessage id="ui-orders.location.quantityElectronic.notMatch" />;
+  return isLocationEresourceQuantityRequired(orderFormat, inventoryType)
+  && allLocationsQuantity !== overallLineQuantity
+    ? <FormattedMessage id="ui-orders.location.quantityElectronic.notMatch" />
+    : undefined;
 };
 
 const parseQuantity = (value) => {
   return value ? Number(value) : 0;
 };
 
+export const isLocationsRequired = (values, valuesAll) => {
+  const orderFormat = get(valuesAll, 'orderFormat');
+  const physicalInventory = get(valuesAll, 'physical.createInventory', '');
+  const eresourceInventory = get(valuesAll, 'eresource.createInventory', '');
+
+  if (
+    (isLocationPhysicalQuantityRequired(orderFormat, physicalInventory)
```

Comment:
something wrong with line breaks, IMHO 
```
if (
  (
    isLocationPhysicalQuantityRequired(orderFormat, physicalInventory) 
    || isLocationEresourceQuantityRequired(orderFormat, eresourceInventory)
  ) && !values.length
) {
```

Reply:
but I suggest to extract booleans to readable vars, so condition could be read easily